client/components/app.jsx

```diff
@@ -17,14 +19,23 @@ class App extends React.Component {
   getAllGrades() {
     fetch('/api/grades')
       .then(res => res.json())
-      .then(data => this.setState({ grades: data }));
+      .then(data => this.setState({ grades: data }))
+      .then(() => this.getAverageGrade());
+  }
 
+  getAverageGrade() {
+    let average = 0;
+    for (let i = 0; i < this.state.grades.length; i++) {
+      average += this.state.grades[i].grade;
+    }
+    average = Math.round(average / this.state.grades.length);
+    this.setState({ avg: average });
```

Comment:
The average grade does not belong in state, since it can just be calculated on-the-fly. This method should just `return` the average grade.